### PR TITLE
[202012] [Mellanox] [SKU] Mellanox-SN4700-V48C32 SKU added

### PR DESCRIPTION
**Pull request by vivekrnv**
Signed-off-by: Vivek Reddy Karri <vkarri@nvidia.com>

#### Why I did it

A new SKU for MSN4700 Platform i.e. **Mellanox-SN4700-V48C32**

#### Requirements:

- **Breakout:** 
      Port 1-24: 2x200G 
      Port 25-32: 4x100G
- **Downlinks:** 48 (1-24)
- **Uplinks:** 32 (31-32)
- **Shared Headroom:** Enabled
- **Over Subscribe Ratio:** 1:8
- **Default Topology:** T1
- **Default Cable Length for T1:** 300m
- **VxLAN source port range set:** No
- **Static Policy Based Hashing Supported:** No

##### Additional Details:
-  **QoS params**: The default ones defined in [qos_config.j2](https://github.com/Azure/sonic-buildimage/blob/master/files/build_templates/qos_config.j2) will be applied
-  **Small Packet Percentage:** Used 50% for traditional buffer model Note: For dynamic model, the value defined in `LOSSLESS_TRAFFIC_PATTERN|AZURE|small_packet_percentage` is used
- Cable Lengths used for generating `buffer_defaults_{t0,t1}.j2` values
```
 {
        "t1": {
            "up": "1500m",
            "down": "300m",
        },
        "t0": {
            "up": "300m",
            "down": "5m",
        }
}
```

#### How to verify it

```
root@r-leopard-58:/home/admin# show int status
  Interface            Lanes    Speed    MTU    FEC    Alias    Vlan    Oper    Admin                                             Type    Asym PFC
-----------  ---------------  -------  -----  -----  -------  ------  ------  -------  -----------------------------------------------  ----------
  Ethernet0          0,1,2,3     200G   9100    N/A    etp1a   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
  Ethernet4          4,5,6,7     200G   9100    N/A    etp1b   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
  Ethernet8        8,9,10,11     200G   9100    N/A    etp2a   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
 Ethernet12      12,13,14,15     200G   9100    N/A    etp2b   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
 Ethernet16      16,17,18,19     200G   9100    N/A    etp3a   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
 Ethernet20      20,21,22,23     200G   9100    N/A    etp3b   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
 Ethernet24      24,25,26,27     200G   9100    N/A    etp4a   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
 Ethernet28      28,29,30,31     200G   9100    N/A    etp4b   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
 Ethernet32      32,33,34,35     200G   9100    N/A    etp5a   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
 Ethernet36      36,37,38,39     200G   9100    N/A    etp5b   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
 Ethernet40      40,41,42,43     200G   9100    N/A    etp6a   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
 Ethernet44      44,45,46,47     200G   9100    N/A    etp6b   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
 Ethernet48      48,49,50,51     200G   9100    N/A    etp7a   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
 Ethernet52      52,53,54,55     200G   9100    N/A    etp7b   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
 Ethernet56      56,57,58,59     200G   9100    N/A    etp8a   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
 Ethernet60      60,61,62,63     200G   9100    N/A    etp8b   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
 Ethernet64      64,65,66,67     200G   9100    N/A    etp9a   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
 Ethernet68      68,69,70,71     200G   9100    N/A    etp9b   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
 Ethernet72      72,73,74,75     200G   9100    N/A   etp10a   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
 Ethernet76      76,77,78,79     200G   9100    N/A   etp10b   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
 Ethernet80      80,81,82,83     200G   9100    N/A   etp11a   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
 Ethernet84      84,85,86,87     200G   9100    N/A   etp11b   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
 Ethernet88      88,89,90,91     200G   9100    N/A   etp12a   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
 Ethernet92      92,93,94,95     200G   9100    N/A   etp12b   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
 Ethernet96      96,97,98,99     200G   9100    N/A   etp13a   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet100  100,101,102,103     200G   9100    N/A   etp13b   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet104  104,105,106,107     200G   9100    N/A   etp14a   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet108  108,109,110,111     200G   9100    N/A   etp14b   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet112  112,113,114,115     200G   9100    N/A   etp15a   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet116  116,117,118,119     200G   9100    N/A   etp15b   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet120  120,121,122,123     200G   9100    N/A   etp16a   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet124  124,125,126,127     200G   9100    N/A   etp16b   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet128  128,129,130,131     200G   9100    N/A   etp17a   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet132  132,133,134,135     200G   9100    N/A   etp17b   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet136  136,137,138,139     200G   9100    N/A   etp18a   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet140  140,141,142,143     200G   9100    N/A   etp18b   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet144  144,145,146,147     200G   9100    N/A   etp19a   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet148  148,149,150,151     200G   9100    N/A   etp19b   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet152  152,153,154,155     200G   9100    N/A   etp20a   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet156  156,157,158,159     200G   9100    N/A   etp20b   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet160  160,161,162,163     200G   9100    N/A   etp21a   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet164  164,165,166,167     200G   9100    N/A   etp21b   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet168  168,169,170,171     200G   9100    N/A   etp22a   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet172  172,173,174,175     200G   9100    N/A   etp22b   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet176  176,177,178,179     200G   9100    N/A   etp23a   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet180  180,181,182,183     200G   9100    N/A   etp23b   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet184  184,185,186,187     200G   9100    N/A   etp24a   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet188  188,189,190,191     200G   9100    N/A   etp24b   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet192          192,193     100G   9100    N/A   etp25a   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet194          194,195     100G   9100    N/A   etp25b   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet196          196,197     100G   9100    N/A   etp25c   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet198          198,199     100G   9100    N/A   etp25d   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet200          200,201     100G   9100    N/A   etp26a   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet202          202,203     100G   9100    N/A   etp26b   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet204          204,205     100G   9100    N/A   etp26c   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet206          206,207     100G   9100    N/A   etp26d   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet208          208,209     100G   9100    N/A   etp27a   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet210          210,211     100G   9100    N/A   etp27b   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet212          212,213     100G   9100    N/A   etp27c   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet214          214,215     100G   9100    N/A   etp27d   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet216          216,217     100G   9100    N/A   etp28a   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet218          218,219     100G   9100    N/A   etp28b   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet220          220,221     100G   9100    N/A   etp28c   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet222          222,223     100G   9100    N/A   etp28d   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet224          224,225     100G   9100    N/A   etp29a   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet226          226,227     100G   9100    N/A   etp29b   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet228          228,229     100G   9100    N/A   etp29c   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet230          230,231     100G   9100    N/A   etp29d   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet232          232,233     100G   9100    N/A   etp30a   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet234          234,235     100G   9100    N/A   etp30b   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet236          236,237     100G   9100    N/A   etp30c   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet238          238,239     100G   9100    N/A   etp30d   trunk      up       up  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
Ethernet240          240,241     100G   9100    N/A   etp31a  routed    down       up                                  QSFP28 or later         N/A
Ethernet242          242,243     100G   9100    N/A   etp31b  routed    down       up                                  QSFP28 or later         N/A
Ethernet244          244,245     100G   9100    N/A   etp31c  routed    down       up                                  QSFP28 or later         N/A
Ethernet246          246,247     100G   9100    N/A   etp31d  routed    down       up                                  QSFP28 or later         N/A
Ethernet248          248,249     100G   9100    N/A   etp32a  routed    down       up                                  QSFP28 or later         N/A
Ethernet250          250,251     100G   9100    N/A   etp32b  routed    down       up                                  QSFP28 or later         N/A
Ethernet252          252,253     100G   9100    N/A   etp32c  routed    down       up                                  QSFP28 or later         N/A
Ethernet254          254,255     100G   9100    N/A   etp32d  routed    down       up                                  QSFP28 or later         N/A
```

```
root@r-leopard-58:/home/admin# show buffer configuration
Pool: egress_lossy_pool
----  --------
mode  dynamic
size  46780416
type  egress
----  --------

Pool: ingress_lossless_pool
----  --------
mode  dynamic
size  46780416
type  ingress
xoff  5091328
----  --------

Pool: egress_lossless_pool
----  --------
mode  dynamic
size  60817392
type  egress
----  --------

Profile: q_lossy_profile
----------  -------------------------------
dynamic_th  3
pool        [BUFFER_POOL|egress_lossy_pool]
size        0
----------  -------------------------------

Profile: pg_lossless_100000_300m_profile
----------  -----------------------------------
dynamic_th  0
pool        [BUFFER_POOL|ingress_lossless_pool]
size        19456
xoff        120832
xon         19456
----------  -----------------------------------

Profile: pg_lossless_200000_300m_profile
----------  -----------------------------------
dynamic_th  0
pool        [BUFFER_POOL|ingress_lossless_pool]
size        19456
xoff        188416
xon         19456
----------  -----------------------------------

Profile: egress_lossless_profile
----------  ----------------------------------
dynamic_th  7
pool        [BUFFER_POOL|egress_lossless_pool]
size        0
----------  ----------------------------------

Profile: ingress_lossy_profile
----------  -----------------------------------
dynamic_th  3
pool        [BUFFER_POOL|ingress_lossless_pool]
size        0
----------  -----------------------------------

Profile: egress_lossy_profile
----------  -------------------------------
dynamic_th  7
pool        [BUFFER_POOL|egress_lossy_pool]
size        9216
----------  -------------------------------

Profile: ingress_lossless_profile
----------  -----------------------------------
dynamic_th  7
pool        [BUFFER_POOL|ingress_lossless_pool]
size        0
----------  -----------------------------------

```

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205

#### A picture of a cute animal (not mandatory but encouraged)
[syslog.gz](https://github.com/sonic-net/sonic-buildimage/files/9264243/syslog.gz)

### CODE REVIEW

**Comment by vivekrnv** [comment]
@stephenxs, @assafho Please review

**Review by dgsudharsan — APPROVED** [review]

File: device/mellanox/x86_64-mlnx_msn4700-r0/Mellanox-SN4700-V48C32/pg_profile_lookup.ini [hunk 1]
```diff
@@ -0,0 +1,41 @@
+##
```
Comment (stephenxs) [inline device/mellanox/x86_64-mlnx_msn4700-r0/Mellanox-SN4700-V48C32/pg_profile_lookup.ini:1]: Let's make this file a symbol link to `../Mellanox-SN4700-C128/pg_profile_lookup.ini`

**Review by stephenxs — CHANGES_REQUESTED** [review]
As comments.
Buffer pool sizes are correct.

**Comment by assafho** [comment]
show buffer configuration is showing the following PG profile:
Profile: pg_lossless_100000_300m_profile

100G is used for uplinks, so it should use the 1500m profile.

**Comment by vivekrnv** [comment]
> show buffer configuration is showing the following PG profile: Profile: pg_lossless_100000_300m_profile
> 
> 100G is used for uplinks, so it should use the 1500m profile.

For the traditional buffer model, the name doesn't matter as we've already calculated size, xoff, xon based on 1500m value in pg_profile_ini. 

The name is derived here based on the default_length we use, which can only be one value and thus set to 300 here. 

@stephenxs correct me if i'm wrong. 

Although, thanks for pointing out.

I'll make some changes to accomodate correct naming scheme. I'll add this to the .j2 file 
```
ports2cable = {
            'torrouter_server'       : '40m',
            'leafrouter_torrouter'   : '300m',
            'spinerouter_leafrouter' : '1500m'
 }
```

But for the correct naming to happen, user has to populate DEVICE_NEIGHBOR_METADATA table for each front panel interface specifying it's own role and neighbors role.

File: device/mellanox/x86_64-mlnx_msn4700-r0/Mellanox-SN4700-V48C32/pg_profile_lookup.ini (hunk 1 above)
Comment (vivekrnv) [reply device/mellanox/x86_64-mlnx_msn4700-r0/Mellanox-SN4700-V48C32/pg_profile_lookup.ini:1]: Will update

Comment (vivekrnv) [reply device/mellanox/x86_64-mlnx_msn4700-r0/Mellanox-SN4700-V48C32/pg_profile_lookup.ini]: Updated

**Review by stephenxs — APPROVED** [review]

**Comment by stephenxs** [comment]
> > show buffer configuration is showing the following PG profile: Profile: pg_lossless_100000_300m_profile
> > 100G is used for uplinks, so it should use the 1500m profile.
> 
> For the traditional buffer model, the name doesn't matter as we've already calculated size, xoff, xon based on 1500m value in pg_profile_ini.
> 
> The name is derived here based on the default_length we use, which can only be one value and thus set to 300 here.
> 
> @stephenxs correct me if i'm wrong.
> 
> Although, thanks for pointing out.
> 
> I'll make some changes to accomodate correct naming scheme. I'll add this to the .j2 file
> 
> ```
> ports2cable = {
>             'torrouter_server'       : '5m',
>             'leafrouter_torrouter'   : '300m',
>             'spinerouter_leafrouter' : '1500m'
>  }
> ```
> 
> But for the correct naming to happen, user has to populate DEVICE_NEIGHBOR_METADATA table for each front panel interface specifying it's own role and neighbors role.

Basically, you are correct except that we need both `DEVICE_NEIGHBOR_METADATA` and `DEVICE_NEIGHBOR`. they should be prepared by the customer during deployment

**Pull request closed without merging**